### DISCUSSION TITLE
AI Tags

### DISCUSSION THREAD

**Discussion by kbatbouta** [CombatExtended-Continued/CombatExtended#890] [General]
## Goal

- To give a clear distinctive `weaponTags` and things alike that facilitate the implementation of AI systems such as smart ammo selection and better weapon switching.

## Suggested weapon tags

These tags are to indicate possible roles of a weapon. 

Note a weapon should have every tag that applies to it since this is intended to discourage or encourage weapon switching, thus if a weapon is suitable for multiple roles then we could just keep using it instead of switching. 

- `CE_AI_Broom` for any weapon effective at close range. (Trench broom)
- `CE_AI_LMG` for any weapon the can be used  for suppression.
- `CE_AI_AR` for any weapon that is effective at medium range. 
- `CE_AI_SNIPER` for any weapon that is effective at medium range to long range with single shots.
- `CE_AI_AOE` to be used to indicate any AOE weapon.
- `CE_AI_Smoke` Given to weapon capable of creating smoke screens.
- `CE_AI_Emp` Given to weapon capable of creating Emp AOE effect.

## Suggested ammo tags

These tags should be used with a new mod extension to `AmmoDef` indicate the type of ammo.

- `CE_Ammo_AOE` used for AOE ammo. (Should be added to all AOE ammo)
- `CE_Ammo_Ball` used for point type ammo.
- `CE_Ammo_AP` used for armor piercing ammo.
- `CE_Ammo_EMP` used for Emp ammo.
- `CE_Ammo_Smoke` used for smoke ammo.

Note: I'm yet to create this new mod extension.

**Comment by ghost** [comment]
#### ` CE_AI_BROOM `

- 1911 *
- Redhawk *
- Mossberg 500
- Saiga 12K
- MAC-10
- UMP45
- Taurus Judge *
- USAS-12
- AKS-74U **
- Charge PDW **

\* Should we include handguns?

#### ` CE_AI_LMG `

- Degtyarev DP
- RPD
- M60
- PKM
- Charge LMG

#### ` CE_AI_AR `

- M16A4
- Charge Rifle
- FN FAL
- AK47
- AK-12
 
#### ` CE_AI_SR `

- PTRS (?)
- Lee Enfield
- M24
- SKS
- SVD
- Tac-50
- Charge Sniper

#### ` CE_AI_AOE `

- M79
- T-9 Launcher
- Triple Rocket
- Doomsday Launcher
- RM2 Flamethrower
- M72 LAW
- RPG
- Milkor MGL

### What to do with grenades?
Frag - ` CE_AI_AOE `
Smoke - ` CE_AI_SMOKE `
EMP - ` CE_AI_EMP `
Flashbang?

**Comment by Breadbox2k19** [comment]
CE_AI_Default: 
For slow firing weapons such as bolt actions, crossbows and musket, they default to the baseline aim-mode as defined in their xml and do not try to switch modes based on distance as they'll just miss everytime.
Also acts as a fail safe for any weapons that doesn't fit the existing catergory.
